pycbc/live/snr_optimizer.py

```diff
@@ -0,0 +1,384 @@
+    opt_opt_group.add_argument('--optimizer',
+        type=str,
+        default='differential_evolution',
+        choices=optimizer_choices,
+        help='The optimizer to use, choose from ' + ', '.join(optimizer_choices))
+
+    # Add the generic options
+    opt_opt_group.add_argument('--include-candidate-in-optimizer',
+        action='store_true',
+        help='Include parameters of the candidate event in the initialized array for the optimizer. '
+             'Only relevant for --optimizer pso or differential_evolution')
+    opt_opt_group.add_argument('--optimizer-seed',
+        type=int,
+        default=42,
+        help='Seed to supply to the random generation of initial array to pass to the optimizer. '
+             'Only relevant for --optimizer pso or differential_evolution')
+
+    # For each optimizer, add the possible options
+    for optimizer, option_subdict in option_dict.items():
+        optimizer_name = optimizer.replace('_','-')
+        for opt_name, opt_help_default in option_subdict.items():
+            option_name = f"--{optimizer_name}-{opt_name}"
+            opt_opt_group.add_argument(option_name,
+                type=float,
+                help=f'Only relevant for --optimizer {optimizer}: ' +
+                     opt_help_default[0] +
+                     f' Default = {opt_help_default[1]}')
+
+
+def check_snr_optimizer_options(args, parser):
+    """
+    Deal with default options and required parameters given optimizer option
+    """
+    options = {}
+    options['differential_evolution'] = [args.differential_evolution_maxiter,
+                                         args.differential_evolution_popsize]
+    options['shgo'] = [args.shgo_samples, args.shgo_iters]
+    options['pso'] = [args.pso_iters, args.pso_particles, args.pso_c1,
+                      args.pso_c2, args.pso_w]
+
+    if args.optimizer == 'pso' and ps == None:
+        parser.error('You need to install pyswarms to use the pso optimizer.')
+
+    # Have any options been given for a different optimizer?
+    # If so, raise a warning
+    for k in options.keys():
+        if args.optimizer == k:
+            continue
+        if any(options[k]):
+            parser.error("Argument has been supplied which is not suitable " +
+                         f"for the optimizer given ({args.optimizer})")
+
+    # Give the arguments the default values according to the dictionary
+    for key, value in option_dict[args.optimizer].items():
+        key_name = f'{args.optimizer}_{key}'
+        if not getattr(args, key_name):
+            setattr(args, key_name, value[1])
+
+def args_to_string(args):
+    """
+    Convert the supplied arguments for SNR optimization config into
+    a string - this is to be used when running subprocesses
+    """
+    argstr = f' --optimizer {args.optimizer} '
+    optimizer_name = args.optimizer.replace('_','-')
+    for opt in option_dict[args.optimizer]:
+        option_fullname = f"--{optimizer_name}-{opt}"
```

Comment:
this is slightly messy in that some of these strings have spaces at the end and/or the beginning and some don't .. can it be redone so that they all have exactly 1 space at the beginning? 
That way it should be easier to check / parse. 

Also this line has double quotes whereas all the others have single .. hopefully this is not for any reason, then it's easier to read with all single ;)